### ISSUE TITLE
Site Editor: Consider adding in descriptions to "Add New Templates" flow

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#59046]
Right now, you can toggle/hide descriptions for templates when in data views, but when you click "Add a new template" there's no description of the template you might add:

![Screenshot 2024-02-14 at 3 51 41 PM](https://github.com/WordPress/gutenberg/assets/26996883/01cc5477-3ebb-4038-82ee-35df1bd27417)

This makes it hard to know at a glance what each template might do. Perhaps when you hover, a tooltip could show? Or we could consider adding in very brief descriptions.

**Comment by t-hamano** [comment]
In #50595, when this modal dialog was implemented, a description existed. After that, it seems that the description  was deleted in #51806.

I'm leaning toward bringing back the description, but the current description is [quite long](https://github.com/WordPress/wordpress-develop/blob/7b24083859c37d8ac75b9549f458bc389610eaef/src/wp-includes/block-template-utils.php#L109-L174), so I think it would be nice to add a short description.

example:

```php
function get_default_block_template_types() {
	$default_template_types = array(
		// ...
		'archive'        => array(
			'title'       => _x( 'All Archives', 'Template name' ),
			'description' => __( 'Displays any archive, including posts by a single author, category, tag, taxonomy, custom post type, and date. This template will serve as a fallback when more specific templates (e.g. Category or Tag) cannot be found.' ),
			'short_description' => __( 'Displays any archive, including posts by a single author, category, tag, taxonomy, custom post type, and date.' ),
		),
		// ...
	);
	// ...
}
```

cc @ntsekouras @richtabor @jameskoster @jasmussen

**Comment by jameskoster** [comment]
That's right, the descriptions made the modal extremely noisy. Short descriptions could be worth a try, but it's not always easy to capture the full template purpose succinctly, especially things like `front-page` and `home`, or generic templates like `archive`.

There could even be a world where this becomes a data view grid with previews. Given blocks like Archive Title
now include contextual placeholders (for instance in the Author archives template it displays "Author name" on the canvas), the previews could be more useful. Then users / platforms could toggle fields like description accordingly.

**Comment by annezazu** [comment]
Agreed completely around contextual placeholders. That's where my brain went too. Right now, with the power of data views there, it would be awesome for this modal to then "respect" the choices set.

**Comment by ntsekouras** [comment]
>There could even be a world where this becomes a data view grid with previews. Given blocks like Archive Title
now include contextual placeholders (for instance in the Author archives template it displays "Author name" on the canvas), the previews could be more useful. Then users / platforms could toggle fields like description accordingly.

This depends on whether a theme:
1. has some basic fallback templates like `archive`
2. in those templates some blocks with contextual placeholders are used

If some of the above are not the case for a theme, we can end up in multiple identical previews (e.g. index) which provide no value.

Additionally, even if the blocks with contextual placeholders exist etc.., are we sure they are easily distinguishable and easily scanned? The previews will not be huge and it depends on so many factors, like layouts, colors etc..

No strong opinions but IMO short descriptions seem to be a better way forward.

**Comment by t-hamano** [comment]
Considering the design improvements made in #51806, what do you think about restoring the description? What are your thoughts on this design? If the description is lengthy, we could potentially add a shorter description option.

| Before | After |
|--------|--------|
| <img width="860" height="724" alt="Image" src="https://github.com/user-attachments/assets/89bbd672-254b-4e5b-a25b-7043a09de5d6" /> | <img width="860" height="730" alt="Image" src="https://github.com/user-attachments/assets/9d032b25-9403-470b-966b-53014f39770a" /> |

**Comment by jasmussen** [comment]
No strong opinion either way. Could we use a help-text font size? 

Not sure if possible, but the main downside is all the buttons get different heights now, not sure if grid allows us a  way to keep that consistent 🤔

**Comment by t-hamano** [comment]
> the main downside is all the buttons get different heights now, not sure if grid allows us a way to keep that consistent

We can apply a minimum height [as before](https://github.com/WordPress/gutenberg/pull/51806/changes#diff-70ab64cce594236859b34c50d62162dbd4ad7b844a0380d57a7ade2d90f11effL151), but it would be impossible to make all grid cells match in height using only CSS.

**Comment by jameskoster** [comment]
A nice middle-ground could be to display descriptions as [InfoTips](https://wordpress.github.io/gutenberg/?path=/story/design-system-components-popover--infotip&globals=dsCornerRadius:subtle) to avoid adding so much noise to the dialog.

**Comment by t-hamano** [comment]
> A nice middle-ground could be to display descriptions as [InfoTips](https://wordpress.github.io/gutenberg/?path=/story/design-system-components-popover--infotip&globals=dsCornerRadius:subtle) to avoid adding so much noise to the dialog.

Does this mean overlapping the trigger button on each template button, like this? I feel like this might be a bit too much 😄

https://github.com/user-attachments/assets/d7d8ef22-ac1a-4635-9d37-2a1227761dc4

By the way, we could also display the description when the template button itself is hovered or focused, but that raises accessibility concerns because mobile devices do not support hover or focus states.

**Comment by jasmussen** [comment]
Also, isn't that a button inside a button? 

I feel like the descriptions are so important that for this modal you see so rarely, it might be okay to have them for each. Are we already using the smallest font size?

**Comment by t-hamano** [comment]
> Also, isn't that a button inside a button?

That is correct, and this is undesirable from an accessibility standpoint.

> I feel like the descriptions are so important that for this modal you see so rarely, it might be okay to have them for each. Are we already using the smallest font size?

I submitted a PR that applies a minimum font size to the description. https://github.com/WordPress/gutenberg/pull/83064

**Comment by jameskoster** [comment]
There are other ways to include an `InfoTip` without nesting inside a button.

**Comment by t-hamano** [comment]
> There are other ways to include an `InfoTip` without nesting inside a button.

@jameskoster A rough sketch or simple mockup would be fine, so I would really appreciate it if you could share your ideal UI. I would be more than happy to try implementing it.

**Comment by jameskoster** [comment]
Your original [mockup](https://github.com/WordPress/gutenberg/issues/59046#issuecomment-5707478758) works pretty well imo. It seems very similar to DataViews grid items where the checkbox and menu trigger overlap an interactive item:

<img width="298" height="404" alt="Image" src="https://github.com/user-attachments/assets/8cc020bc-e777-4297-85ec-b7aa5b1904ab" />

Maybe we could do something similar here to avoid the nesting issue? The only other tweak I'd consider is a `max-width` on the info-tip to optimise readability, that should arguably built into the component via `size` prop though (like `Dialog`).
